File: src/api-client/mcp.ts
import { invoke } from "@tauri-apps/api/core";

export interface McpServerSummary {
  name: string;
  transport_type: string;
  status: string; // "connected" | "error" | "disconnected"
  tools_count: number;
  tools: string[];
  error?: string | null;
}

export const getMcpServerPaths = (): Promise<string[]> =>
  invoke("get_mcp_server_paths");

export const checkMcpConfigStatus = (agentId: string, workspaceRoot: string | null): Promise<boolean> =>
  invoke("check_mcp_config_status", { agentId, workspaceRoot });

export const reloadMcpServers = (workspaceRoot: string): Promise<McpServerSummary[]> =>
  invoke("llm2_mcp_reload", { workspaceRoot, workspace_root: workspaceRoot });

export const listMcpServers = (): Promise<McpServerSummary[]> =>
  invoke("llm2_mcp_list_servers");

export const disconnectMcpServer = (serverName: string): Promise<McpServerSummary[]> =>
  invoke("llm2_mcp_disconnect_server", { serverName, server_name: serverName });

export const connectMcpServer = (
  workspaceRoot: string,
  serverName: string
): Promise<McpServerSummary[]> =>
  invoke("llm2_mcp_connect_server", {
    workspaceRoot,
    workspace_root: workspaceRoot,
    serverName,
    server_name: serverName,
  });

export const clearAllMcpServers = (): Promise<void> =>
  invoke("llm2_mcp_clear_all");

File: src-tauri/src/llm2/commands/mcp.rs
use tauri::State;

use crate::mcp_client::{McpClientManager, McpServerSummary};

#[tauri::command]
pub async fn llm2_mcp_reload(
  workspace_root: String,
  state: State<'_, McpClientManager>,
) -> Result<Vec<McpServerSummary>, String> {
  state.reload_for_workspace(&workspace_root).await
}

#[tauri::command]
pub async fn llm2_mcp_list_servers(
  state: State<'_, McpClientManager>,
) -> Result<Vec<McpServerSummary>, String> {
  Ok(state.list_servers().await)
}

#[tauri::command]
pub async fn llm2_mcp_disconnect_server(
  server_name: String,
  state: State<'_, McpClientManager>,
) -> Result<Vec<McpServerSummary>, String> {
  state.disconnect_server(&server_name).await
}

#[tauri::command]
pub async fn llm2_mcp_connect_server(
  workspace_root: String,
  server_name: String,
  state: State<'_, McpClientManager>,
) -> Result<Vec<McpServerSummary>, String> {
  state.connect_server(&workspace_root, &server_name).await
}

#[tauri::command]
pub async fn llm2_mcp_clear_all(
  state: State<'_, McpClientManager>,
) -> Result<(), String> {
  state.clear_all().await;
  Ok(())
}
